docs/30-customize-cluster.md

```diff
@@ -1,19 +1,122 @@
-# Running Commands when the Cluster Starts
+# Customizing the cluster
+
+There are several ways to control what gets deployed to a cluster. The most flexible and powerful method is to manage the docker conatiner image that is used to provision the cluster. By default doAzureParallel uses containers to provision the R environement. Anything within the container will be available on all nodes in the cluster. The default container used in doAzureParallel is [rocker/tidyverse:latest](https://hub.docker.com/r/rocker/tidyverse/) developed and maintained by the rocker organization.
+
+## Modifying the default docker container
+
+Speficying a docker container is done by updating your cluster.json file. Simply adding a reference to the docker container in the cluster.json file 'containerImage' property will cause all new clusters to deploy that conatiner to your cluster. doAzureParallel will use the version of R specified in the container.
+
+```json
+{
+  "name": "myPool",
+  "vmSize": "Standard\_F2",
+  "maxTasksPerNode": 1,
+  "poolSize": {
+    "dedicatedNodes": {
+      "min": 0,
+      "max": 0
+    },
+    "lowPriorityNodes": {
+      "min": 1,
+      "max": 1
+    },
+    "autoscaleFormula": "QUEUE"
+  },
+  "containerImage": "rocker/tidyverse:3.4.1",
+  "rPackages": {
+    "cran": [],
+    "github": [],
+    "bioconductor": [],
+    "githubAuthenticationToken": ""
+  },
+  "commandLine": []
+}
+```
+
+Note: \_If no 'containerImage' property is set, rocker/tidyverse:latest will be used. This usually points to one of the latest versions of R.\_
+
+### Building your own container
+
+Building your own container gives you the flexibility to package any specific requirements, packages or data you require for running your workloads. We recommend using a debian based OS such as debian or ubuntu to build your containers and pointing to where R is in the final CMD command. For example:
```

Comment:
Should we also include in the docs that we recommend install libcurl/httr for installing CRAN packages and github packages?

Reply:
@brnleehng - Do you mean as part of the docker image?

Reply:
Yeah, I guess people might not need any CRAN packages. So I guess it's fine unless people request it

Reply:
i think we should call out building on top one of our 'supported' images with `FROM` because that makes it much more likely that it will work and may reduce our support cost in the future